evmd/app.go

```diff
@@ -765,10 +767,21 @@ func NewExampleApp(
 	// set the EVM priority nonce mempool
 	// If you wish to use the noop mempool, remove this codeblock
 	if evmtypes.GetChainConfig() != nil {
-		// TODO: Get the actual block gas limit from consensus parameters
+		// Get the block gas limit from genesis file
+		blockGasLimit := getBlockGasLimitFromGenesis(appOpts, logger)
+
+		// Get MinGasPrices from app configuration
+		minGasPricesStr := cast.ToString(appOpts.Get(sdkserver.FlagMinGasPrices))
+		minGasPrices, err := sdk.ParseDecCoins(minGasPricesStr)
+		if err != nil {
+			logger.With("error", err).Info("failed to parse min gas prices, using empty DecCoins")
+			minGasPrices = sdk.DecCoins{}
+		}
```

Comment:
All of this logic is in the `emvd` package.  Could we move some portion of it to the `evm` library and export from there?